src/main/java/io/jenkins/plugins/sprp/YAML_FlowDefinition.java

```diff
@@ -60,17 +49,81 @@ public Object readResolve() {
     }
 
     public YAML_FlowDefinition(String scriptPath) {
-        this.scriptPath = "Jenkinsfile.yaml";
+        this.scriptPath = scriptPath;
     }
 
     @Override
     public FlowExecution create(FlowExecutionOwner owner, TaskListener listener,
                                           List<? extends Action> actions) throws Exception {
-        String script = "node {\n" +
-                        "    stage('Example') {\n" +
-                        "            echo 'Done Cloning'\n" +
-                        "    }\n" +
-                        "}";
+        Queue.Executable exec = owner.getExecutable();
+        if (!(exec instanceof WorkflowRun)) {
+            throw new IllegalStateException("inappropriat   e context");
+        }
+
+//        WorkflowRun build = (WorkflowRun) exec;
+//        WorkflowJob job = build.getParent();
+
+//        String script = ""  +
+//                "pipeline {\n" +
+//                "\tagent {\n" +
+//                "\t\tnode {\n" +
+//                "\t\t\tlabel 'my-defined-label'\n" +
+//                "\t\t\tcustomWorkspace '/some/other/path'\n" +
+//                "\t\t}\n" +
+//                "\t\tdocker 'maven:3-alpine'\n" +
+//                "\t}\n" +
+//                "\tstages {\n" +
+//                "\t\tstage('Example') {\n" +
+//                "\t\t\tsteps {\n" +
+//                "\t\t\t\tcheckout scm \n" +
+//                "\t\t\t\techo 'Hello World'\n" +
+//                "\t\t\t\tarchiveArtifacts includes: './Jenkinsfile.yaml', excludes: 'sdf, iiit, cxc'\n" +
+//                "\t\t\t\tscript {\n" +
+//                "\t\t\t\t\tif (isUnix()) {\n" +
+//                "\t\t\t\t\t\tsh 'echo \"This is UNIX\"'\n" +
+//                "\t\t\t\t\t} else {\n" +
+//                "\t\t\t\t\t\tsh 'echo \"This is not UNIX\"'\n" +
+//                "\t\t\t\t\t}\n" +
+//                "\t\t\t\t}\n" +
+//                "\t\t\t}\n" +
+//                "\t\t}\n" +
+//                "\t}\n" +
+//                "}";
+//        script = new YamlToPipeline().generatePipeline();
+
+//        File file = new File("/mnt/CC0091D90091CB3A/workspace/OpenSource/jenkinsOrg/simple-pull-request-job-plugin/work/workspace");
+//
+//
+//        GitOperations gitOperations = new GitOperations(file, listener,
+//                build.getCharacteristicEnvVars(), "https://github.com/gautamabhishek46/dummy");
+
+//        listener.getLogger().println("Credential id = " + this.credentialId);
+//        listener.getLogger().println("Credential id = " + new YAML_BranchProjectFactory().getCredentialsId());
+//        listener.getLogger().println("script path = " + new YAML_BranchProjectFactory().getScriptPath());
+
+
+        YamlToPipeline y = new YamlToPipeline();
+        String script = y.generatePipeline(this.scriptPath, listener);
+
+//        StandardCredentials c = CredentialsMatchers.firstOrNull(
```

Comment:
If you leave the commented code, please add TODO comments, which explain what you are going to do with them (with JIRA links if appropriate)